### ISSUE TITLE
Do you plan to keep supporting ManifestV2 ?

### ISSUE THREAD

**Issue by patientx** [Alex313031/thorium#438]
Google is going to end mv2 support in the very near future, this especially gonna hit adblockers hard. Does this project going to keep supporting mv2 (don't know if its possible at all) ? 

Keep up the good work.

**Comment by fernandofig** [comment]
See #79 

TL;DR: Alex intends to keep ManifestV2 support through an enterprise policy hack. But that will only extend the deadline on which ManifestV2 is supported on Thorium - at some point Google will completely remove the code supporting ManifestV2, and at that point Alex says he'll not be able to support a fork to add back and maintain the removed code.

Now, piggybacking into your question, there's something that I haven't seen discussed anywhere so far: I believe Brave is the only browser that has so far commited to keep ManifestV2 support after Google removes it from Chromium. I worry that I don't see no other browser vendors coming forward with that commitment (though I'd be happy to be corrected if I'm wrong). I fear we need more Chromium forks commiting to keeping ManifestV2 to make that viable.

But my real concern here is how would that look for end users. For the sake of argument, let's say hypothetically that in 6 months from now we have a few chromium forks working together to keep the ManifestV2 code working past Google pulling the plug: how would that look like for end users? What I mean is, the most common way to install extensions is through the web store, but Google has already said they will no longer allow ManifestV2 extensions on the store sometime early next year, so how will users install v2 extensions? Presumably manually, but that wouldn't be great UX. If (e.g.) Brave really intends to keep Mv2 support without sacrificing user friendlyness, the logical step would be to create a new webstore that would allow Mv2 extensions. That's why ideally I think there should be more vendors commited and working together to keep Mv2 extension support up, so eventually a webstore supporting Mv2 could be built that would be shared by all the forks supporting Mv2. But I see no conversations about that, and I fear that Mv2 will just fall out of favor simply due to lack of an easy way to install and use these extensions.

The other side of the problem too would be extension developers. Specifically uBO, are we to expect that they will keep supporting two variants of the extension, one for v2 and another for v3?

Of course, I hope any of this don't happen, and maybe it's too early for these concerns, but I'm just putting this out there as food for thought...

**Comment by patientx** [comment]
Very informative reply , thanks a lot. Their (uborigin) v3 extension it seems works at a limited capacity, cannot add own filters, cannot update as you wish etc. The ability to "edit" the page contents is especially important for people with slower net speeds, I use kiwi on android with desktop extensions and that makes life on the net livable with slow speeds and with this advertisement hell. In the end we don't have any power to change things , just hoping some kind of solution can be found before google completely pulls the plug.

**Comment by fernandofig** [comment]
> Their (uborigin) v3 extension it seems works at a limited capacity, cannot add own filters, cannot update as you wish etc.

How well ManifestV3-compatible uBO works is beside the point. My question is, would uBO be willing to maintain two variants of their extension, of which one of them (manifestV2) would see very limited use once Google disables Mv2 on Chromium - that is, of course, if there's no organized, consortium-like effort to maintain Mv2 on a relevant enough amount of Chromium forks, which so far doesn't look like to be the case, but I guess we'll see.

In fact, in a worst case scenario, since Mv3 uBO is so crippled, it may be possible that uBO devs may just completely drop Chromium support entirely (for Mv2 and Mv3 alike), and just stick to Firefox, though I find it unlikely, IMHO.

**Comment by peteratf** [comment]
> Brave really intends to keep Mv2 support without sacrificing user friendlyness, the logical step would be to create a new webstore that would allow Mv2 extensions. That's why ideally I think there should be more vendors commited and 

brave just says or moreso doesnt say things you want to hear/not hear. always been like this. they dont know what google is planning thus dont know what can and cant be possible and how things will work. no one does. try not to panic over this just see what happens lol. theres nothing you can do now anyways. what if they choose not to go ahead with removal. what if they provide a new tech that shows no ads.
